### PR TITLE
feat: Admit new input: packageName

### PR DESCRIPTION
**Pull request by jrmanes**
## Overview

Hello team!

Some context here.

Sometimes is possible that we have multiples `Dockerfiles` in the same repo (monorepo approach for example), and we have the need to specify a different package in our registry, this feature allow us to specify via github inputs the registry name where we'll publish the container.

ℹ️ **Keeping by default, the one that we were using before, the repository name.**

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Already tested: [here](https://github.com/jrmanes/.github/blob/main/.github/workflows/reusable_dockerfile_pipeline.yml) with [this](https://github.com/jrmanes/docker-cicd/blob/main/.github/workflows/common_dockerfile_pipeline.yml)
  - [x] [Actions executed](https://github.com/jrmanes/docker-cicd/actions/runs/4231927614)

### CODE REVIEW

**Review by Bidon15 — APPROVED** [review]

File: .github/workflows/reusable_dockerfile_pipeline.yml [hunk 1]
```diff
@@ -26,6 +31,11 @@ jobs:
           echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
           # yamllint disable
           echo "IMAGE_NAME=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
+          # here we validate if we have specified a different package name in
+          # the inputs, if so, we change the package to it.
+          if [[ ${{ inputs.packageName }} != ${{ github.repository}} ]];then
+            echo "IMAGE_NAME=$(echo ${{ github.repository_owner }}/${{ inputs.packageName }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
```
Comment (smuu) [inline .github/workflows/reusable_dockerfile_pipeline.yml:37]: Users could add other characters than the ones supported by a container registry. Maybe check with a short regex like `([A-Za-z0-9\-]+)` with a helpful output.

File: .github/workflows/reusable_dockerfile_pipeline.yml [hunk 2]
```diff
@@ -75,6 +85,11 @@ jobs:
...
```
Comment (smuu) [inline .github/workflows/reusable_dockerfile_pipeline.yml:91]: Code duplication: Is it possible to put that in one action and use the output from both actions? This way, we only need to change one place to add functionality.

**Review by smuu — CHANGES_REQUESTED** [review]

File: .github/workflows/reusable_dockerfile_pipeline.yml (hunk 2 above)
Comment (jrmanes) [reply .github/workflows/reusable_dockerfile_pipeline.yml:91]: I couldn't share these variables between jobs very clear... 
I saw the `outputs` but looks a little bit tricky... like [this](https://stackoverflow.com/a/63485275), not sure if it worth it

**Review by smuu — DISMISSED** [review]

**Comment by jrmanes** [comment]
hello @Bidon15 @smuu 

i've added a previous step for preparing the env vars and reusing them later, it's already tested and works:
[example here](https://github.com/jrmanes/docker-cicd/actions/runs/4232429138)

File: .github/workflows/reusable_dockerfile_pipeline.yml [hunk 3]
```diff
@@ -53,8 +83,8 @@ jobs:
 
   docker-build:
     runs-on: "ubuntu-latest"
-    # wait until the security scanner will be done
-    needs: docker-security
+    # wait until the jobs are finished.
+    needs: ["prepare-env","docker-security"]
```
Comment (Bidon15) [inline .github/workflows/reusable_dockerfile_pipeline.yml:87]: this should do the trick I suppose
```suggestion
    needs: ["prepare-env", "docker-security"]
```

**Review by Bidon15 — COMMENTED** [review]
One nit and LFG 

Apart from that - we can consider moving to this style in the future 
https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action

File: .github/workflows/reusable_dockerfile_pipeline.yml (hunk 1 above)
Comment (jrmanes) [reply .github/workflows/reusable_dockerfile_pipeline.yml:51]: good point!
already done and tested!

https://github.com/jrmanes/docker-cicd/actions/runs/4232871935/jobs/7353129308


cc: @Bidon15

**Comment by jrmanes** [comment]
> One nit and LFG
> 
> Apart from that - we can consider moving to this style in the future https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action

Interesting one! 📝 
as this pipeline starts growing, maybe this could be the next(future) iteration

**Comment by Bidon15** [comment]
> > One nit and LFG
> > Apart from that - we can consider moving to this style in the future https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action
> 
> Interesting one! 📝 as this pipeline starts growing, maybe this could be the next(future) iteration

https://github.com/celestiaorg/.github/issues/37

**Review by Bidon15 — APPROVED** [review]

**Review by smuu — APPROVED** [review]
:rocket:

**Pull request merged**